modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector.tsx

```diff
@@ -54,14 +63,29 @@ const noop = () => {};
 
 const TriggerButton = React.forwardRef(
 	(
-		{displayType, selectedItem, small, ...props}: TriggerButtonProps,
+		{
+			displayType,
+			selectedItem,
+			small,
+			triggerRef,
+			...props
+		}: TriggerButtonProps,
 		ref: Ref<HTMLButtonElement>
 	) => {
 		const ariaLabelButton = sub(
 			Liferay.Language.get('select-a-language.-current-language-x'),
 			selectedItem.displayName
 		);
 
+		useEffect(() => {
+			if (ref && triggerRef) {
+
+				// @ts-ignore
+
+				triggerRef.current = ref.current;
+			}
+		}, [ref, triggerRef]);
```

Comment:
Why do we need `triggerRef` here? This assignment can be in parent, using `ref` exposed by `forwardRef`. 

In general, passing in a ref as a prop feels wrong to me, an anti-pattern that messes up separation of concerns. I cannot think of a case where it would be needed, and not possible to do with either normal prop or a callback.

Reply:
Thank you for your review! I'm curious why you think this is an anti-pattern :sweat_smile: . If using React 16.2 as is our case, the [React docs](https://legacy.reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components) mention the custom ref property as a more flexible approach to forwardRef:

> If you use React 16.2 or lower, or if you need more flexibility than provided by ref forwarding, you can use this alternative approach and explicitly pass a ref as a prop with a different name.

Anyway, we have changed the way translations are calculated in our modules, so we have reconsidered this approach. We think it is more convenient for now create a new prop `onTriggerClick` and pass a function that is executed only when the selector is opened. We have updated this commits (https://github.com/liferay-frontend/liferay-portal/pull/3966/commits/05f1a10c66bada69c1865cad578421a9185b42a6 and https://github.com/liferay-frontend/liferay-portal/pull/3966/commits/9e5c9c4b12be23a0b8ba19d876b969c5abd74aaa). What do you think? :slightly_smiling_face:

Reply:
> I'm curious why you think this is an anti-pattern

My thinking here goes not only to passing ref as a prop, but using `forwardRef` as well. It's about separation of concerns. As they say in docs you linked:
> This is generally not recommended because it breaks component encapsulation, but it can occasionally be useful for triggering focus or measuring the size or position of a child DOM node.

So, it's not a recommended pattern (an anti-pattern), but sometimes necessary. In the case of the PR, if was not necessary, the new solution based on a callback is much cleaner!